docs/architecture/0004-eq-author-permissions-model.md

```diff
@@ -0,0 +1,61 @@
+# 4. EQ Author Permissions Model
+
+Date: 2019-03-11
+
+## Status
+
+Accepted
+
+## Context
+
+As part of the work to move Author into a live environment we have decided to implement a permissions model so that we are able to better control what users are able to do to questionnaires in the system.
+
+## Decision
+
+We're proposing to implement a permissions model for Author with the following design.
+
+### Querying a questionnaire
+
+When the user logs in:
+
+1. Get the current user Id from the token.
+1. Loop through the questionnaire list.
+1. For each questionnaire check that the user has questionnaire read permission by looking it up in questionnaire list table.
+1. Show the questionnaire in the list if the user has read permissions, if the user doesn't have read permissions do not show questionnaire in the list
+
+### Data tables
+
+#### Users
+
+List of users
+
+| ID  | User Name |
+| --- | --------- |
+| 1   | User1     |
+| 2   | User2     |
+
+<br>
+
+#### Questionnaire List
+
+List of questionnaires
+
+| ID  | Title | CreatedAt     | UpdatedAt     | Createdby | Readlist       | Writelist      |
+| --- | ----- | ------------- | ------------- | --------- | -------------- | -------------- |
+| 1   | OPN   | 1532563200000 | 1532563200000 | User1     | [User1, User2] | [User1, User2] |
+| 2   | UKIS  | 1532563200000 | 1532563200000 | User2     | [User1, User2] | [User1, User2] |
+
+<br>
+
+#### Questionnaires Versions
+
+List of questionnaires by versions
+
+| ID  | Title | CreatedAt     | UpdatedAt     | CreatedBy | LegalBasis           | Metadata             | ... |
+| --- | ----- | ------------- | ------------- | --------- | -------------------- | -------------------- | --- |
+| 1   | OPN   | 1532563200000 | 1532563200000 | User1     | StatisticsOfTradeAct | {"alias":"Trad As... | ... |
+| 2   | UKIS  | 1532563200000 | 1532563200000 | User2     | StatisticsOfTradeAct | {"alias":"Trad As... | ... |
+
+## Consequences
+
+No risks this is a requirement of making Author a production system.
```

Comment:
The only risk that we might have to deal with is initially, to maintain the status quo, we would give everyone read and write permissions on all questionnaires. Over time, we will seek to selectively remove permissions on questionnaires to protect people's work. And empower users to decide for themselves who should have access.